pkg/api/types.go

```diff
@@ -3312,6 +3314,31 @@ const (
 	TLSCertKey = "tls.crt"
 	// TLSPrivateKeyKey is the key for the private key field in a TLS secret.
 	TLSPrivateKeyKey = "tls.key"
+
+	// SecretBootstrapToken is used during the automated bootstrap process (first
+	// implemented by kubeadm). It stores tokens that are used to sign well known
+	// ConfigMaps. They may also eventually be used for authentication.
+	SecretTypeBootstrapToken SecretType = "bootstrap.kubernetes.io/token"
```

Comment:
Have we ever added a non-core secret type?  I would say this doesn't need to be in this code at all, but should be in bootstrap specific code.  Especially if you're claiming bootstrap.kubernetes.io as a subdomain for this work.